### PR TITLE
add new tools dnabot v 3.1.0

### PR DESCRIPTION
**Pull request by kenza12**
Add new tool **dnabot v 3.1.0** : DNA assembly using BASIC on OpenTrons (https://github.com/BASIC-DNA-ASSEMBLY/DNA-BOT)
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/dnabot/dnabot.xml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+                <option value="A4" >A4</option>
+                <option value="A5" >A5</option>
+                <option value="A6" >A6</option>
+                <option value="A7" >A7</option>
+                <option value="A8" >A8</option>
+                <option value="A9" >A9</option>
+                <option value="A10" >A10</option>
+                <option value="A11" selected="true">A11</option>
+            </param>
+            <param name="soc_column" type="select" label="Column coordinate for SOC">
```
Comment (bernt-matthias) [inline tools/dnabot/dnabot.xml:40]: Can you explain `SOC` somewhere?

File: tools/dnabot/dnabot.xml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        --construct_path '$construct_file'
+        #set files = '" "'.join([str($file) for $file in $plate_files])
+        --source_paths "${files}"
+        --etoh_well '$adv.etoh_well'
+        --soc_column '$adv.soc_column'
+        --output_dir 'output'
+        && tar -cvf '$dnabot_scripts' 'output'
+    ]]></command>
+    <inputs>
+        <param name="construct_file" type="data" format="csv" label="Source Construct" />
```
Comment (bernt-matthias) [inline tools/dnabot/dnabot.xml:24]: `tabular` would be preferable.

File: tools/dnabot/dnabot.xml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                <option value="8" >8</option>
+                <option value="9" >9</option>
+                <option value="10" >10</option>
+                <option value="11" >11</option>
+                <option value="12" >12</option>
+            </param>
+        </section>
+    </inputs>
+    <outputs>
+        <data name="dnabot_scripts" format="tar" label="${tool.name} scripts" />
```
Comment (bernt-matthias) [inline tools/dnabot/dnabot.xml:57]: Just keep the default label if there is only one output.

Comment (bernt-matthias) [reply tools/dnabot/dnabot.xml:57]: Can this tar file be used by downstream tools?

File: tools/dnabot/dnabot.xml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+* **construct_path**: (string) Construct CSV file.
+* **source_paths**: (string) Source CSV files.
+* **etoh_well**: (string) Well coordinate for Ethanol. Default: A11.
+* **soc_column**: (integer) Column coordinate for SOC. Default: 1.
+* **template_dir**: (string) Template directory. Default: "template_ot2_scripts" located next to the present script.
+
+Ouput
+-----
+
+* **output_dir**: (string) Output directory. Default: same directory than the one containing the "construct_path" file.
```
[the remark this answers is not in the document]
Comment (kenza12) [reply tools/dnabot/dnabot.xml]: The help section is updated and help fields are added. Thanks.

File: tools/dnabot/dnabot.xml (hunk 3 above)
Comment (bernt-matthias) [reply tools/dnabot/dnabot.xml]: Sounds like it could be an optional output.

File: tools/dnabot/dnabot.xml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    </inputs>
+    <outputs>
+        <data name="dnabot_scripts" format="tar" />
+    </outputs>
+    <tests>
+        <test>
+        <!-- test 1: check if identical outputs are produced with compress option -->
+            <param name="construct_file" value="constructs.csv" />
+            <param name="plate_files" value="user_parts_coords.csv,linker_parts_coords.csv"/>
+            <output name="dnabot_scripts" file="dnabot_scripts.tar" compare="sim_size" decompress="True"/>
```
Comment (bernt-matthias) [inline tools/dnabot/dnabot.xml:64]:
```suggestion
            <output name="dnabot_scripts" ftype="tar">
                <assert_contents>
                    <has_size value="TODO" delta="10000"/>
                </assert_contents>
            </output>
```

You may use this instead and remove the tar file from the test-data folder. Just insert the size of the file in bytes for `TODO`. You may also change `delta` (10000 is the default for `sim_size` which you used before).

**Review by bernt-matthias — COMMENTED** [review]
Just one more minor thing and maybe explain the abbreviation SOC somewhere...

File: tools/dnabot/dnabot.xml (hunk 3 above)
Comment (bernt-matthias) [reply tools/dnabot/dnabot.xml]: Forget my previous comment :)

**Comment by bernt-matthias** [comment]
For the help section you should always target users that have no clue about the application and domain (like me :) ). Since I still have no idea what the tool is doing I guess that there is still room for improvement. 

So if you have the time to extend the help a bit it would be great, if not we can also merge as is and hope that it may be extended in the future.

File: tools/dnabot/dnabot.xml [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    </inputs>
+    <outputs>
+        <data name="dnabot_scripts" format="tar" />
+    </outputs>
+    <tests>
+        <test>
+        <!-- test 1: check if identical outputs are produced with compress option -->
+            <param name="construct_file" value="constructs.csv" />
+            <param name="plate_files" value="user_parts_coords.csv,linker_parts_coords.csv"/>
+            <output name="dnabot_scripts" ftype="tar">
```
Comment (bernt-matthias) [inline tools/dnabot/dnabot.xml:64]: The test file `tools/dnabot/test-data/dnabot_scripts.tar` may now be removed.

Comment (kenza12) [reply tools/dnabot/dnabot.xml]: Hi @bernt-matthias , I already deleted this test file as you can see here : https://github.com/galaxyproject/tools-iuc/pull/4458/files/a8078b2b6be95db42e6cba686926d7a813255e3a#diff-7cda21ab4bff8ecd00ba36f4bb2f9d845ff5f46062c614aa9374187508f23220

**Comment by kenza12** [comment]
> For the help section you should always target users that have no clue about the application and domain (like me :) ). Since I still have no idea what the tool is doing I guess that there is still room for improvement.
> 
> So if you have the time to extend the help a bit it would be great, if not we can also merge as is and hope that it may be extended in the future.

I added more information in help section including a figure. I hope it's clearer now. We will see in the future if we can add more information. We're also trying to create a tutorial in Galaxy Training where this tool is included.

Thank you for your work and your advice.

File: tools/dnabot/dnabot.xml (hunk 5 above)
Comment (kenza12) [reply tools/dnabot/dnabot.xml]: Hi, I just added delta to the correct field, sorry for the previous mistakes.

**Comment by bgruening** [comment]
Thanks @kenza12 :)

If you like: https://galaxyproject.eu/posts/2020/08/22/three-steps-to-galaxify-your-tool/

**Comment by kenza12** [comment]
Hi @bgruening , I was looking for dnabot tool in the toolshed from IUC but I can't find it. Is it normal ? Thanks.

**Comment by nsoranzo** [comment]
@kenza12 Not sure what went wrong at the time since too much time has passed, but I've now manually uploaded dnabot to the ToolShed. Thanks for reporting the issue!

**Pull request merged**